test/unittest/unittest_nntrainer_models.cpp

```diff
@@ -751,6 +751,90 @@ INI conv_none_loss_validate(
   }
 );
 
+INI pooling_max_same_padding(
+  "pooling_max_same_padding",
+  {
+    nn_base + "learning_rate=0.1 | optimizer=sgd | loss=cross | batch_size=3",
+        I("input") + input_base + "input_shape=2:5:3",
```

Comment:
It would be good to have these tests for 2:3:5 to ensure the working along both height and width dimensions. These extra test cases can be skipped for global pooling versions.

Reply:
Added, also testing for the multistride